core/iwasm/fast-jit/fe/jit_emit_memory.c

```diff
@@ -5,18 +5,317 @@
+    if (!offset1)
+        goto fail;
+#endif
+
+    /* ---------- seek ---------- */
+    memory_data = get_memory_data_reg(cc->jit_frame, mem_idx);
+    maddr = jit_cc_new_reg_ptr(cc);
+    GEN_INSN(ADD, maddr, memory_data, offset1);
+
+    return maddr;
+fail:
+    return 0;
+}
 
 bool
 jit_compile_op_i32_load(JitCompContext *cc, uint32 align, uint32 offset,
                         uint32 bytes, bool sign, bool atomic)
 {
+    JitReg addr, maddr, value;
+
+    POP_I32(addr);
+
+    maddr = check_and_seek(cc, addr, offset, bytes);
+    if (!maddr) {
+        goto fail;
+    }
+
+    value = jit_cc_new_reg_I32(cc);
+    switch (bytes) {
+        case 1:
+        {
+            if (sign) {
+                JitReg tmp1 = jit_cc_new_reg_I32(cc);
+                JitReg tmp2 = jit_cc_new_reg_I32(cc);
+                GEN_INSN(LDI8, tmp1, maddr, NEW_CONST(I32, 0));
+                GEN_INSN(SHL, tmp2, tmp1, NEW_CONST(I32, 24));
+                GEN_INSN(SHRS, value, tmp2, NEW_CONST(I32, 24));
+            }
+            else {
+                JitReg tmp1 = jit_cc_new_reg_I32(cc);
+                GEN_INSN(LDU8, tmp1, maddr, NEW_CONST(I32, 0));
+                GEN_INSN(AND, value, tmp1, NEW_CONST(I32, 0x000000FF));
+            }
+            break;
+        }
+        case 2:
+        {
+            if (sign) {
+                JitReg tmp1 = jit_cc_new_reg_I32(cc);
+                JitReg tmp2 = jit_cc_new_reg_I32(cc);
+                GEN_INSN(LDI16, tmp1, maddr, NEW_CONST(I32, 0));
+                GEN_INSN(SHL, tmp2, tmp1, NEW_CONST(I32, 16));
+                GEN_INSN(SHRS, value, tmp2, NEW_CONST(I32, 16));
+            }
+            else {
+                JitReg tmp1 = jit_cc_new_reg_I32(cc);
+                GEN_INSN(LDU16, tmp1, maddr, NEW_CONST(I32, 0));
+                GEN_INSN(AND, value, tmp1, NEW_CONST(I32, 0x0000FFFF));
+            }
+            break;
+        }
+        case 4:
+        {
+            if (sign) {
+                GEN_INSN(LDI32, value, maddr, NEW_CONST(I32, 0));
+            }
+            else {
+                GEN_INSN(LDU32, value, maddr, NEW_CONST(I32, 0));
+            }
+            break;
+        }
+        default:
+        {
+            bh_assert(0);
+            goto fail;
+        }
+    }
+
+    PUSH_I32(value);
+    return true;
+fail:
     return false;
 }
 
 bool
 jit_compile_op_i64_load(JitCompContext *cc, uint32 align, uint32 offset,
                         uint32 bytes, bool sign, bool atomic)
 {
+    JitReg addr, maddr, value;
+
+    POP_I32(addr);
+
+    maddr = check_and_seek(cc, addr, offset, bytes);
+    if (!maddr) {
+        goto fail;
+    }
+
+    value = jit_cc_new_reg_I64(cc);
+    switch (bytes) {
+        case 1:
+        {
+            if (sign) {
+                JitReg tmp1 = jit_cc_new_reg_I64(cc);
+                JitReg tmp2 = jit_cc_new_reg_I64(cc);
+                GEN_INSN(LDI8, tmp1, maddr, NEW_CONST(I32, 0));
```

Comment:
Same as above, using LDI8/LDU8/LDI16/LDU16/LDI32/LDU32 should be OK, it is the codegen's responsibility to signed/unsigned extend the memory data to i32/i64.